catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -224,6 +229,16 @@ private String getSubjectEmail() {
     return SubjectUtils.getAttribute(SecurityUtils.getSubject(), Constants.ROLES_CLAIM_URI);
   }
 
+  // TODO: DDF-4249 to refactor this logic for PreQueryPlugin Access Control
+  private boolean isElevatedUser(List<String> subjectRoles) {
+    List<String> systemUserList =
+        StringUtils.isNotEmpty(accessControlSecurityConfiguration.getSystemUserAttributeValue())
+            ? Collections.singletonList(
+                accessControlSecurityConfiguration.getSystemUserAttributeValue())
+            : Collections.emptyList();
+    return !Collections.disjoint(subjectRoles, systemUserList);
```

Comment:
❗️ This is the wrong way of doing this. You cannot make decisions in an endpoint, you need to create a set of permissions and then check whether or not the subject is permitted. That is the way we do it everywhere else in the system.

https://github.com/codice/ddf/blob/master/platform/security/filter/security-filter-authorization/src/main/java/org/codice/ddf/security/filter/authorization/AuthorizationFilter.java#L84

https://github.com/codice/ddf/blob/master/catalog/security/catalog-security-filter/src/main/java/ddf/catalog/security/filter/plugin/FilterPlugin.java#L349

Reply:
This again is not about policy enforcement.  Those plugins continue to serve as the enforcer.  This is about an optimization to get back a whole lot less that they will reject.  I can accept that this is how enforcement must be done, but I think we should be able to write clients that will make requests that should provide results that are expected to be a complete list of acceptable results, and allow the policy enforcement plugins to do their job from a security perspective.

I do not believe we can be an effective system if we continue to allow the post query policy to effectively be our query constraints.  We need to leverage the database and index capabilities.

Reply:
This just isn't the right way to check whether or not a Subject is allowed to do something. I'm not saying that you need to use the post query filter method, I AM saying that you need to do `subject.isPermitted(...)`